pkg/list/list_test.go

```diff
@@ -1,17 +1,47 @@
-package list_test
+package list
```

Comment:
I am using the `_test` package to be sure about the public interface, is there any reason to rename this?

Reply:
best practices on package names. You can make a root folder named `./test` and add a test there testing the public API (basically what you do in the `./main.go`)